service/controller/resource/monitoring/scrapeconfigs/resource.go

```diff
@@ -54,6 +55,22 @@ type TemplateData struct {
 	CAPICluster               bool
 }
 
+func getAPIAuthenticationMechanism(cluster metav1.Object, clusterType string, config Config) (string, error) {
```

Comment:
Move this function to a shared place instead of duplicating it.

Reply:
Done